libtrip-notification/src/main/java/com/mapbox/navigation/trip/notification/MapboxTripNotification.kt

```diff
@@ -312,7 +313,9 @@ class MapboxTripNotification constructor(
                 }
             }
             setFreeDriveMode(false)
-        } ?: setFreeDriveMode(true)
+        } else {
+            setFreeDriveMode(true)
```

Comment:
This bears a question - how are we going to transition between notification free-drive/active-guidance states if route progress will not be delivered in free-drive?

Reply:
This is a great question. Thought about this and I believe it's time for `MapboxTripNotification` to be aware of the `NavigationSession.State`. Any other ideas?

Reply:
Route being `null` in a progress update has been away we've recommended other users to recognize free-drive and active-guidance. Is it time to expose and document the whole nav state classes with the drawbacks we discussed in the past?

Reply:
> Is it time to expose and document the whole nav state classes with the drawbacks we discussed in the past?

For now, I think we can keep those internal and rely on them in the `MapboxTripNotification` implementation somehow. I'm 🤔 about it and will push a draft 🔜 In any case, as always ideas / feedback welcome 🙏

Reply:
It turned out that wasn't necessary with the approach followed. Let me know what you think.